packages/fluentui/react/src/components/Table/TableCell.tsx

```diff
@@ -1,108 +1,135 @@
+    className: TableCell.className,
+    mapPropsToStyles: () => ({
+      truncateContent
     }),
-    content: customPropTypes.every([
-      customPropTypes.disallow(['children']),
-      PropTypes.oneOfType([PropTypes.arrayOf(customPropTypes.nodeContent), customPropTypes.nodeContent])
-    ]),
-    truncateContent: PropTypes.bool
-  };
-
-  static defaultProps = {
-    accessibility: tableCellBehavior as Accessibility
-  };
-
-  cellRef = React.createRef<HTMLElement>();
-
-  actionHandlers = {
-    focusCell: e => {
-      e.preventDefault();
-      this.cellRef.current.focus();
-    },
-    performClick: e => {
-      this.handleClick(e);
-    }
-  };
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables
+    }),
+    rtl: context.rtl
+  });
 
-  handleClick = (e: React.SyntheticEvent) => {
+  const handleClick = (e: React.SyntheticEvent) => {
     if (e.currentTarget === e.target) {
-      _.invoke(this.props, 'onClick', e, this.props);
+      _.invoke(props, 'onClick', e, props);
       e.preventDefault();
     }
   };
 
-  renderComponent({ accessibility, ElementType, styles, classes, unhandledProps }: RenderResultConfig<any>): React.ReactNode {
-    const { children, content } = this.props;
-    const hasChildren = childrenExist(children);
-
-    return (
-      <Ref innerRef={this.cellRef}>
-        <ElementType
-          className={classes.root}
-          {...accessibility.attributes.root}
-          {...unhandledProps}
-          {...applyAccessibilityKeyHandlers(accessibility.keyHandlers.root, unhandledProps)}
-        >
-          {hasChildren
-            ? children
-            : Box.create(content, {
-                defaultProps: () => ({ as: 'div', styles: styles.content })
-              })}
-        </ElementType>
-      </Ref>
-    );
-  }
-}
+  const tableCell = (
+    <Ref innerRef={cellRef}>
+      <ElementType
+        {...getA11yProps('root', {
+          className: classes.root,
+          onClick: handleClick,
+          ...unhandledProps
+        })}
+      >
+        {hasChildren
+          ? children
+          : Box.create(content, {
+              defaultProps: () => ({ as: 'div', styles: resolvedStyles.content })
+            })}
+      </ElementType>
+    </Ref>
+  );
+  setEnd();
+  return tableCell;
+};
+
+TableCell.displayName = 'TableCell';
+
+TableCell.className = 'ui-table__cell';
+
+TableCell.slotClassNames = {
+  content: `${TableCell.className}__content`
+};
+
+TableCell.propTypes = {
+  ...commonPropTypes.createCommon({
+    content: false
+  }),
+  content: customPropTypes.every([
+    customPropTypes.disallow(['children']),
+    PropTypes.oneOfType([PropTypes.arrayOf(customPropTypes.nodeContent), customPropTypes.nodeContent])
+  ]),
+  truncateContent: PropTypes.bool
+};
+
+TableCell.handledProps = Object.keys(TableCell.propTypes) as any;
+
+TableCell.defaultProps = {
+  accessibility: tableCellBehavior as Accessibility
```

Comment:
```suggestion
  accessibility: tableCellBehavior
```

Is not needed anymore